### DISCUSSION TITLE
How to publish and view hls stream..??

### DISCUSSION THREAD

**Discussion by daltaniyang** [bluenviron/mediamtx#3477] [Questions]
### Question

--  index.m3u8, file_001.ts file_002.ts file_003.ts ....

What is the command line to publish hls stream and view it in browser..?

ffmpeg -i "index.m3u8" -c:v copy -f hls -hls_time 10 -hls_flags delete_segments http://127.0.0.1:8888/live/stream.m3u8

Is the above command line syntax correct?
It seems like something is going on, but I can't check it in the browser.

“stream not found, retrying in some seconds."

You can only see the message above.

**Comment by moenodedev** [comment]
Does http://127.0.0.1:8888/yourpathname work as the destination? The same URL should be viewable in the browser too.

**Comment by nvtienlg** [comment]
Hello,

I have same question. 

here is part of config file
```
paths:
  ~^transcoded:
    runOnReady:

  all_others:
    runOnReady: >
      ffmpeg -i rtsp://localhost:$RTSP_PORT/$MTX_PATH
      -pix_fmt yuv420p -preset ultrafast -profile:v main -level:v 4.0 -max_muxing_queue_size 2048
      -c:v libx264 -b:v 1200k -c:a libfdk_aac -b:a 64k -ar 48000 -ac 2
      -f rtsp rtsp://localhost:$RTSP_PORT/transcoded$MTX_PATH
    runOnReadyRestart: yes
```

publish on `http://localhost:8889/teststream/publish`
and view on `http://localhost:8888/teststream`
is working

but i expect that hls url: `http://localhost:8888/transcodedteststream/index.m3u8` is working every with all browsers (since it transcoded), but it doesn't. It still use same codecs as source (av01 in my case), not h264

is my config correct? if not then how to transcode the incomming stream to hls with h264 codecs ?

Thank you

**Reply by aler9** [reply]
this is unrelated from this discussion. Open a dedicated discussion and provide more details including server logs.

**Comment by aler9** [comment]
Hello, we don't support publishing streams through HLS. Use another protocol, for instance RTSP:

```
ffmpeg -re -i "index.m3u8" -c:v copy -f rtsp http://127.0.0.1:8554/live
```

Once the stream is inside the server, you will be able to read it with HLS:

```
http://127.0.0.1:8888/live
```

[a comment by an automated account is not shown]
